sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -241,9 +245,79 @@ protected internal string GetSessionFilePath()
 
             string fileName = $"{name}{version}{async}.json";
 
-            return Path.Combine(
+            var repoRoot = GetRepoRoot();
+
+            // this needs to be updated to purely relative to repo root
+            var result = Path.Combine(
                 GetSessionFileDirectory(),
                 fileName);
+
+            return Regex.Replace(result.Replace(repoRoot, String.Empty), @"^[\\/]*", string.Empty);
+        }
+
+        private class DirectoryEvaluation
+        {
+            public bool IsRoot;
+            public bool IsGitRoot;
+            public bool AssetsJsonPresent;
+        }
+
+        private DirectoryEvaluation EvaluateDirectory(string directoryPath)
+        {
+            var assetsJsonLocation = Path.Combine(directoryPath, AssetsJson);
+            var gitLocation = Path.Combine(directoryPath, ".git");
+
+            return new DirectoryEvaluation()
+            {
+                AssetsJsonPresent = File.Exists(assetsJsonLocation),
+                IsGitRoot = Directory.Exists(gitLocation),
+                IsRoot = new DirectoryInfo(directoryPath).Parent == null
+            };
+        }
+
+        public string GetRepoRoot()
+        {
+            var path = GetSessionFileDirectory();
+
+            while (true)
+            {
+                var evaluation = EvaluateDirectory(path);
+
+                if (evaluation.IsGitRoot)
+                {
+                    return path;
+                }
+                else if (evaluation.IsRoot)
+                {
+                    // talk to JoshLove about what exception should actually be thrown here when this isn't a proto
+                    throw new Exception("Search for assets.json failed. Reached system root without finding one.");
```

Comment:
The failure here is that we reached the root without finding a git root, not that we didn't find assets.json I thought.

Reply:
Not even present anymore after my latest commit. Just ripped the entire thing out 👍